model/OAuthClient.php

```diff
@@ -240,7 +272,9 @@ protected function getTokenStorage()
                 'An oauth connection requires the option "' . self::OPTION_TOKEN_STORAGE . '" to store the token'
             );
         }
-        return \common_persistence_Persistence::getPersistence($this->getOption(self::OPTION_TOKEN_STORAGE));
+        return $this->getServiceLocator()
```

Comment:
This will not work, on act CatEngine, the method `getServiceLocator` it's returning null